internal/receiver/awscontainerinsightreceiver/k8sapiserver/k8sapiserver.go

```diff
@@ -0,0 +1,210 @@
+package k8sapiserver
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	common "github.com/aws-observability/aws-otel-collector/internal/aws/containerinsightcommon"
+	"github.com/aws-observability/aws-otel-collector/internal/aws/k8scommon/k8sclient"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog"
+)
+
+const (
+	lockName = "aoc-clusterleader"
+)
+
+type K8sAPIServer struct {
+	nodeName string //get the value from downward API
+	logger   *zap.Logger
+	cancel   context.CancelFunc
+	leading  bool
+}
+
+func New(logger *zap.Logger) *K8sAPIServer {
+	k := &K8sAPIServer{
+		logger: logger,
+	}
+
+	if err := k.start(); err != nil {
+		k.logger.Warn("Fail to start k8sapiserver", zap.Error(err))
+		return nil
+	}
+
+	return k
+}
+
+func (k *K8sAPIServer) GetMetrics() []pdata.Metrics {
+	var result []pdata.Metrics
+	if k.leading {
+		log.Printf("D! collect data from K8s API Server...")
+		timestamp := strconv.FormatInt(time.Now().UnixNano(), 10)
```

Comment:
in OTel, we use `timestampMs` as variable name for timestamp in MS.  :)

Reply:
OK. I will change to use `timestampNs`